SqlPersistence/Timeout/SequentialGuid.cs

```diff
@@ -0,0 +1,33 @@
+namespace NServiceBus.Persistence.Sql
+{
+    using System;
+    using System.Threading;
+
+    // From https://github.com/aspnet/EntityFramework/blob/dev/src/Microsoft.EntityFrameworkCore/ValueGeneration/SequentialGuidValueGenerator.cs
+    static class SequentialGuid
+    {
+        static long counter = DateTime.UtcNow.Ticks;
```

Comment:
:trollface: Well, we can't just use the `Win32` call because of portability. Should we call `YAGNI` on it and revisit it if we actually experience conflicts?